namer/core/src/main/scala/io/buoyant/hostport.scala

```diff
@@ -9,15 +9,20 @@ import com.twitter.finagle.Path
  * work in Paths as-is, due to bracket characters).
  */
 private object HostColonPort {
-  object PortStr {
-    def unapply(s: String): Option[Int] = {
-      val port = try s.toInt catch { case _: java.lang.NumberFormatException => -1 }
-      if (0 < port && port < math.pow(2, 16)) Some(port) else None
-    }
-  }
+  /**
+    * regex for capturing strings that conform to the definition
+    * of a "label" in RFCs 1035 and 1123, used for the `port` part
+    * of a [[HostColonPort]].
+    *
+    * this is based on the `DNS_LABEL` regex in Kubernetes:
+    * https://github.com/kubernetes/kubernetes/blob/master/pkg/api/types.go#L40-L43
+    */
+  private[this] val DnsLabel  = """([a-z0-9][-a-z0-9]*[a-z0-9]?)""".r
 
-  def unapply(s: String): Option[(String, Int)] = s.split(":") match {
-    case Array(host, PortStr(port)) => Some((host, port))
+  def unapply(s: String): Option[(String, String)] = s.split(":") match {
+    case Array(host, DnsLabel(port))
+      // DNS labels may not exceed 63 characters in length
+      if port.length <= 63 => Some((host, port))
     case _ => None
   }
 }
```

Comment:
Done 6460c2e2d3f0dffc9a32298e3aaa16ff1932d017 – I think this should be merge-able whenever?